### PR TITLE
Remove old files after tablet is successfully loaded

### PR DESCRIPTION
**Pull request by yiguolei**
1. check and call init automatically when call load
2. return error if convert meets error
3. remove old files when convert and load tablet finished

### CODE REVIEW

File: be/src/olap/tablet_manager.cpp [hunk 1]
```diff
@@ -1086,8 +1090,9 @@ OLAPStatus TabletManager::_drop_tablet_directly_unlocked(
 
     TabletSharedPtr dropped_tablet = _get_tablet_with_no_lock(tablet_id, schema_hash);
     if (dropped_tablet.get() == NULL) {
```
Comment (kangpinghuang) [inline be/src/olap/tablet_manager.cpp]: dropped_tablet == nullptr

File: be/src/olap/data_dir.h [hunk 2]
```diff
@@ -123,7 +123,7 @@ class DataDir {
     Status _write_cluster_id_to_path(const std::string& path, int32_t cluster_id); 
     OLAPStatus _clean_unfinished_converting_data();
     OLAPStatus _convert_old_tablet();
-    OLAPStatus _remove_old_meta_and_files();
+    OLAPStatus _remove_old_meta_and_files(std::set<int64_t>& tablet_ids);
```
Comment (kangpinghuang) [inline be/src/olap/data_dir.h]:
```suggestion
    OLAPStatus _remove_old_meta_and_files(const std::set<int64_t>& tablet_ids);
```

**Review by kangpinghuang — CHANGES_REQUESTED** [review]

**Review by chaoyli — CHANGES_REQUESTED** [review]

File: be/src/olap/data_dir.cpp [hunk 3]
```diff
@@ -604,18 +605,24 @@ OLAPStatus DataDir::_convert_old_tablet() {
...
         LOG(INFO) << "successfully convert old tablet, data dir: " << _path;
     }
     return OLAP_SUCCESS;
 }
 
-OLAPStatus DataDir::_remove_old_meta_and_files() {
+OLAPStatus DataDir::_remove_old_meta_and_files(std::set<int64_t>& tablet_ids) {
     // clean old meta(olap header message) 
-    auto clean_old_meta_func = [this](long tablet_id,
+    auto clean_old_meta_func = [this, &tablet_ids](long tablet_id,
```
[the remark this answers is not in the document]
Comment (yiguolei) [reply be/src/olap/data_dir.cpp]: done

File: be/src/olap/data_dir.cpp [hunk 4]
```diff
@@ -604,18 +605,24 @@ OLAPStatus DataDir::_convert_old_tablet() {
...
         long schema_hash, const std::string& value) -> bool {
+        if (tablet_ids.count(tablet_id) == 0) {
```
[the remark this answers is not in the document]
Comment (yiguolei) [reply be/src/olap/data_dir.cpp]: done

**Review by chaoyli — APPROVED** [review]

**Pull request merged**